src/components/reader-page-image-list.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import React, { Component, Fragment } from 'react';
+import type { Page } from 'poketo';
+import mostCommon from 'array-most-common';
+import type { PageDimensions } from '../types';
+import ReaderPageImage from '../components/reader-page-image';
+
+type PageListProps = {
+  pages: Page[],
+};
+
+type PageListState = {
+  defaultPageWidth?: number,
+  defaultPageHeight?: number,
+};
+
+function serializeDimension(dimension: PageDimensions) {
```

Comment:
We serialize and unserialize the dimensions here because the `array-most-common` package doesn't work with objects. It doesn't seem to have a noticeable impact on performance though, so it's all good.